src/Persistent/Idiom.hs

```diff
@@ -49,28 +49,51 @@ quorum idea = case idea ^. ideaLocation . ideaLocationSpace of
     SchoolSpace  -> getDb dbSchoolQuorum
     ClassSpace _ -> getDb dbClassQuorum
 
--- | Return the current system time with the day set to the date on which refinement phases opened
+-- | Return the current system time with the day set to the date on which phases opened
 -- today end.  When running the phase change trigger at midnight, find all dates that lie in the
 -- past.
-phaseEndRefinement :: PersistM m => m Timestamp
-phaseEndRefinement = do
+phaseEnd :: PersistM m => Int -> m Timestamp
+phaseEnd days = do
     Timestamp timestamp <- getCurrentTimestamp
-    DurationDays (days :: Int) <- getDb dbElaborationDuration
     let day' :: Integer = toModifiedJulianDay (utctDay timestamp) + fromIntegral days
     return . Timestamp $ timestamp { utctDay = ModifiedJulianDay day' }
 
--- | Returns the Just phase of an idea if the idea is assocaited with a topic, Nothing
+phaseEndRefinement :: PersistM m => m Timestamp
+phaseEndRefinement = do
+    DurationDays (days :: Int) <- getDb dbElaborationDuration
+    phaseEnd days
+
+phaseEndVote :: PersistM m => m Timestamp
+phaseEndVote = do
+    DurationDays (days :: Int) <- getDb dbVoteDuration
+    phaseEnd days
+
+
+-- | Returns the Just topic of an idea if the idea is assocaited with a topic, Nothing
 -- if the idea is a wild idea, or throws an error if the topic is missing.
-ideaPhase :: PersistM m => Idea -> m (Maybe Phase)
-ideaPhase idea = case idea ^. ideaLocation of
+ideaTopic :: PersistM m => Idea -> m (Maybe Topic)
+ideaTopic idea = case idea ^. ideaLocation of
```

Comment:
we really need to clean up Persistent.Api, and now Persistent.Idiom.  there are a lot of functions similar to this one in Api already.  (to be done later)